### PR TITLE
Hai 430 hankkeet kartalla rajapinta

### PR DESCRIPTION
**Pull request by umeetiusbaar**
* /hankkeet?geometry=true (possibly with other search parameters as well) includes geometry in results
* Added Hanke.geometriat
* Added HankeSearch.geometry
* NOTICE TODO in HankeController.getHankeList for moving search logic into service instead of controller - I think it would be better in service (checking what kind of search it is and returning results accordingly).

### CODE REVIEW

File: services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/HankeControllerITests.kt [hunk 1]
```diff
@@ -53,7 +59,7 @@ class HankeControllerITests(@Autowired val mockMvc: MockMvc) {
     }
 
     @Test
-    fun `When calling get without parameters then return all Hanke data`() {
+    fun `When calling get without parameters then return all Hanke data without geometry`() {
```
Comment (tintotti) [inline services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/HankeControllerITests.kt:62]: Good addition for naming!

File: services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeController.kt [hunk 2]
```diff
@@ -80,23 +75,9 @@ class HankeController(@Autowired private val hankeService: HankeService) {
...
-            ResponseEntity.status(HttpStatus.OK).body(hankeList)
-
-        } catch (e: Exception) {
-            logger.error(e) {
-                HankeError.HAI1004.toString()
-            }
-            ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(HankeError.HAI1004)
+    private fun includeGeometry(hankeList: List<Hanke>) {
+        hankeList.forEach { hanke ->
+            hanke.geometriat = hankeGeometriatService.loadGeometriat(hanke.hankeTunnus!!)
```
Comment (tintotti) [inline services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeController.kt:80]: Did you consider getting all geometrys in one call? Now service is called separately for all hanke. If this is thought process then OK :)

File: services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeController.kt [hunk 3]
```diff
@@ -47,31 +51,22 @@ class HankeController(@Autowired private val hankeService: HankeService) {
...
-
-        if (hankeSearch == null || hankeSearch.periodBegin == null || hankeSearch.periodEnd == null ) {
-            return getAllHankeItems()
-        }
         return try {
-            //  Get all hanke datas within time period (= either or both of alkuPvm and loppuPvm are inside the requested period)
-            // TODO: user token  from front?
-            //  TODO: do we limit result for user own hanke?
-            val hankeList = hankeService.loadAllHankeBetweenDates(hankeSearch.periodBegin!!, hankeSearch.periodEnd!!)
+            // TODO it might be better to move this logic with different search types into service and keep controller as clean and lean as possible
```
Comment (tintotti) [inline services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeController.kt:55]: That is not bad idea. Should have thought that when controller started to get more complex.

**Review by tintotti — APPROVED** [review]

File: services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeController.kt (hunk 2 above)
Comment (umeetiusbaar) [reply services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeController.kt]: I did not consider changing into that. But I did change what you suggested elsewhere about loading geometries without also loading the Hanke first (as was done for the geometry loading with only hankeTunnus before).
Loading geometries for multiple Hanke at once would need more changes into database queries etc. Something for future development perhaps...

File: services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/geometria/HankeGeometriatServiceImpl.kt [hunk 4]
```diff
@@ -3,6 +3,7 @@ package fi.hel.haitaton.hanke.geometria
 import fi.hel.haitaton.hanke.HankeNotFoundException
 import fi.hel.haitaton.hanke.HankeRepository
 import fi.hel.haitaton.hanke.TZ_UTC
+import fi.hel.haitaton.hanke.domain.Hanke
```
Comment (muhassin) [inline services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/geometria/HankeGeometriatServiceImpl.kt:6]: unused import?

Comment (umeetiusbaar) [reply services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/geometria/HankeGeometriatServiceImpl.kt]: fixed

**Review by muhassin — APPROVED** [review]

**Pull request merged**